src/components/FilterBar/FilterBar.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import style from './FilterBar.module.css';
+import TuneIcon from '@mui/icons-material/Tune';
+import GridViewRoundedIcon from '@mui/icons-material/GridViewRounded';
+import ViewDayOutlinedIcon from '@mui/icons-material/ViewDayOutlined';
+import BasicSelect from '../BasicSelect/BasicSelect';
+import { TextField } from '@mui/material';
+import IconButton from '@mui/material/IconButton';
+import useStyles from './filterBarStyle';
+import Menu from '@mui/material/Menu';
+import FilterMenu from '../FilterMenu/FilterMenu';
+import { useState,useContext} from 'react';
+import {dataContext,viewContext} from '../ShoppingProducts/ShoppingProducts';
+import { Box } from '@mui/material';
+import Typography from '@mui/material/Typography';
+
+const FilterBar = () => {
+
+  const [anchorEl, setAnchorEl] = useState(null);
+  const open = Boolean(anchorEl);
+  const [data, setData] = useContext(dataContext);
+  const dataLimited=[...data];
+  const [limit,setLimit]=useState(30);
+  const [isViewSolo,setIsViewSolo]=useContext(viewContext);
+  
+  const handleClick = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleChange = (event) => {
+    setLimit(event.target.value);
+  };
+const handleLimiting=(event)=>{
+  // if(event.target.value<data.length)
+  setData(dataLimited.slice(0, event.target.value));
+  console.log(dataLimited.slice(0, event.target.value));
+}
+
+const handleToggleToSolo = () => {
+  if(!isViewSolo)
+   setIsViewSolo(!isViewSolo);
+};
+const handleToggleToGrid = () => {
+  if(isViewSolo)
+   setIsViewSolo(!isViewSolo);
+};
+  const classes = useStyles();
+  return (
+    <Box className={style.filterBar}>
+        <Box className={style.leftSide}>
+          <Box className={style.iconsBtn}> 
+            <IconButton onClick={handleClick}><TuneIcon className={style.icon}/><h4>Filter</h4></IconButton>
+            <Menu id="basic-menu" className={classes.menu} anchorEl={anchorEl} open={open} onClose={handleClose} MenuListProps={{'aria-labelledby': 'basic-button', }}>
+              <FilterMenu />
+            </Menu>
+            <IconButton onClick={handleToggleToGrid}><GridViewRoundedIcon className={style.icon}/></IconButton>
```

Comment:
please add a Tooltip from MUI to IconButtons to let the user know what this button is for